src/methods/to.js

```diff
@@ -669,6 +669,20 @@ export class ToImplementation {
         let resultType = result.getType();
 
         // ii. If Type(result) is not Object, return result.
+        if (resultType === Value) {
+          invariant(result instanceof AbstractValue);
+          let error = new CompilerDiagnostic(
+            `${name} might return either an object or primitive`,
+            realm.currentLocation,
+            "PP0028",
+            "RecoverableError"
+          );
+          if (realm.handleError(error) === "Recover") {
+            // A well behaved toString or valueOf is assumed to return a primitive.
+            return result;
```

Comment:
This may be wrong since result need not be primitive, in which case the second method may have to be called if name is the name of the first method. Since it is hard to tell at this point, and not possible to emit an AbstractValue that will call both values (if need be), the best thing to do here is to give a FatalError and let the caller figure out how to delay things until runtime.

Reply:
What is the intention of "Recover"? It is somewhat saying that we assume it is "well behaved" but what does that mean? Must it have a specific meaning or can it be broader in some sense of what we expect will happen?

Reply:
The reporter of an error can mark its severity as "RecoverableError", which means that it is offering to make an assumption and fix things up so that analysis can continue. If the client is interested in continuing the analysis, it returns "Recover" from errorHandler and undertakes to collect, organize and display multiple errors without duplication.

In some situations, typically when there is speculative evaluation, all kinds of errors are downgraded to warnings and the result of errorHandler is ignored.